cmake/find/ccache.cmake

```diff
@@ -49,11 +44,14 @@ if (CCACHE_FOUND AND NOT COMPILER_MATCHES_CCACHE)
       # - 4.0+ will ignore SOURCE_DATE_EPOCH environment variable.
       if (CCACHE_VERSION VERSION_GREATER_EQUAL "4.2")
          message(STATUS "ccache is 4.2+ no quirks for SOURCE_DATE_EPOCH required")
+         set(LAUNCHER ${CCACHE_FOUND})
```

Comment:
I think this needs to be moved to line 34, otherwise LAUNCHER is undefined for ccache < 4.0

Reply:
Yep, you are right, nice catch!
Can you submit a PR?

Reply:
Hope this works https://github.com/ClickHouse/ClickHouse/pull/33306